src/irr/asset/CCPUMeshPacker.h

```diff
@@ -46,85 +67,243 @@ PackedMeshBufferData CCPUMeshPacker<MDIStructType>::alloc(const ICPUMeshBuffer c
 
 	//validation
 	//TODO: remove this condition
+	for(auto it = begin; it != end; it++)
 	{
-		auto* pipeline = meshBuffer->getPipeline();
+		auto* pipeline = (*it)->getPipeline();
 
-		auto a = meshBuffer->getIndexBufferBinding().buffer;
+		auto a = (*it)->getIndexBufferBinding()->buffer;
 
-		if (meshBuffer->getIndexBufferBinding().buffer.get() == nullptr ||
+		if ((*it)->getIndexBufferBinding()->buffer.get() == nullptr ||
 			pipeline->getPrimitiveAssemblyParams().primitiveType != EPT_TRIANGLE_LIST)
 		{
 			_IRR_DEBUG_BREAK_IF(true);
-			return invalidPackedMeshBufferData;
+			return invalidReservedAllocationMeshBuffers;
 		}
 	}
 
 	//validation
-	const auto& mbVtxInputParams = meshBuffer->getPipeline()->getVertexInputParams();
-	for (uint16_t attrBit = 0x0001, location = 0; location < 16; attrBit <<= 1, location++)
+	for (auto it = begin; it != end; it++)
 	{
-		if (!(attrBit & mbVtxInputParams.enabledAttribFlags))
-			continue;
+		const auto& mbVtxInputParams = (*it)->getPipeline()->getVertexInputParams();
+		for (uint16_t attrBit = 0x0001, location = 0; location < SVertexInputParams::MAX_ATTR_BUF_BINDING_COUNT; attrBit <<= 1, location++)
+		{
+			if (!(attrBit & mbVtxInputParams.enabledAttribFlags))
+				continue;
 
-		//assert((attrBit & m_outVtxInputParams.enabledAttribFlags));
+			//assert((attrBit & m_outVtxInputParams.enabledAttribFlags));
 
-		if (mbVtxInputParams.attributes[location].format != m_outVtxInputParams.attributes[location].format ||
-			mbVtxInputParams.bindings[mbVtxInputParams.attributes[location].binding].inputRate != m_outVtxInputParams.bindings[location].inputRate)
-		{
-			_IRR_DEBUG_BREAK_IF(true);
-			return invalidPackedMeshBufferData;
+			if (mbVtxInputParams.attributes[location].format != m_outVtxInputParams.attributes[location].format ||
+				mbVtxInputParams.bindings[mbVtxInputParams.attributes[location].binding].inputRate != m_outVtxInputParams.bindings[location].inputRate)
+			{
+				_IRR_DEBUG_BREAK_IF(true);
+				return invalidReservedAllocationMeshBuffers;
+			}
 		}
 	}
+	
+	const size_t idxCnt = std::accumulate(begin, end, 0ull, [](size_t init, ICPUMeshBuffer* mb) { return init + mb->getIndexCount(); });
+	const size_t vtxCnt = std::accumulate(begin, end, 0ull, [](size_t init, ICPUMeshBuffer* mb) { return init + mb->calcVertexCount(); });
+
+	const uint32_t minIdxCntPerPatch = m_minTriangleCountPerMDIData * 3;
+	const uint32_t possibleMDIStructsNeededCnt = core::roundUp<uint32_t>(idxCnt, minIdxCntPerPatch);
```

Comment:
no thats not the correct counting, correct is:
```
possibleMDIStructsNeededCnt = 0u;
for (meshbuffer : meshBufferList)
{
    possibleMDIStructsNeededCnt += (meshbuffer->getIndexCount()+minIdxCntPerPatch-1)/minIdxCntPerPatch;
}
```
P.S. Also `roundUp` does not perform the division